lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -600,6 +600,37 @@ SILCombiner::optimizeConcatenationOfStringLiterals(ApplyInst *AI) {
   return tryToConcatenateStrings(AI, Builder);
 }
 
+/// Determine the pattern for global_addr..
+/// %3 = global_addr @$P : $*SomeP
+/// %4 = init_existential_addr %3 : $*SomeP, $SomeC
+/// %5 = alloc_ref $SomeC
+/// store %5 to %4 : $*SomeC
+static SILValue findInitExistentialFromGlobalAddr(GlobalAddrInst *GAI,
+                  CopyAddrInst *CAI) {
+  /// Check for a single InitExistential.
+  InitExistentialAddrInst *SingleIE = nullptr;
+  auto *BB = CAI->getParent();
+  for (auto Use : GAI->getUses()) {
+    auto *User = Use->getUser();
+    if (auto *InitExistential = dyn_cast<InitExistentialAddrInst>(User)) {
+      /// Simple dominance check: both InitExistential and CAI are in 
+      /// the same basic block and not more than one InitExistential 
+      /// occurs before CAI.
+      auto *IEBB = InitExistential->getParent();
+      if (BB == IEBB &&
+            (std::find_if(InitExistential->getIterator(), BB->end(), 
+              [&CAI] (const SILInstruction &I) -> bool {
+                return CAI == &I;
+              }) != BB->end())) {
+        if (SingleIE) return SILValue();
+        SingleIE = InitExistential;
+      }
+    }
+  }
+  if (!SingleIE) return SILValue();
```

Comment:
``SILValue()`` == ``SILValue(nullptr);``

So you should be able to just return SingleIE.